### PR TITLE
Make GDL support a feature

### LINKED ISSUE #22: Hide `GDL` support behind a feature flag
as suggested by @knutwalker

### PR DESCRIPTION
**Pull request by s1ck**
fixes #22

### CODE REVIEW

**Comment by s1ck** [comment]
@knutwalker maybe you wanna have a look :)

**Review by knutwalker — APPROVED** [review]
Nice, thank you!

One small addition I have is to use the doc_cfg feature. Since we're already on nightly, this is rather straightforward:


```
diff --git src/lib.rs src/lib.rs
index 664262a..085eace 100644
--- src/lib.rs
+++ src/lib.rs
@@ -4,6 +4,7 @@
 #![feature(maybe_uninit_slice)]
 #![feature(step_trait)]
 #![feature(new_uninit)]
+#![feature(doc_cfg)]
 #![allow(dead_code)]
```

Which allows us to use the following

```
diff --git src/builder.rs src/builder.rs
index d1669c2..3788e17 100644
--- src/builder.rs
+++ src/builder.rs
@@ -43,6 +43,7 @@ where
 }

 #[cfg(feature = "gdl")]
+#[doc(cfg(feature = "gdl"))]
 pub struct FromGdlString<NI>
 where
     NI: Idx,
```

(and all the other places with the feature attribute as well)

This gives us labels on the documentation, indicating the required feature:

image

image

File: src/input/mod.rs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 pub mod binary;
 pub mod dotgraph;
 pub mod edgelist;
+#[cfg(feature = "gdl")]
```
Comment (s1ck) [inline src/input/mod.rs:4]: @knutwalker I guess we can keep that one here instead of the `docs_cfg`?

Comment (s1ck) [reply src/input/mod.rs:4]: nvm, we need that one to flag all the `From` impls in there, too

**Pull request merged**